service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -305,7 +305,7 @@ public Map<UUID, WsmIamRole> listWorkspaceIdsAndHighestRoles(
           // Skip workspaces with no roles. (That means there's a role this WSM doesn't know
           // about.)
           if (highestRole.isPresent()) {
-            if (WsmIamRole.roleAtLeastAsHighAs(minimumHighestRoleFromRequest, highestRole.get())) {
+            if (minimumHighestRoleFromRequest.roleAtLeastAsHighAs(highestRole.get())) {
               workspacesAndRoles.put(workspaceId, highestRole.get());
             }
```

Comment:
Another minor improvement here; you can use `Optional.map()` to combine `isPresent()` with `get()`.
```
WsmIamRole.getHighestRole(workspaceId, roles).map(higestRole -> { ... });
```
In general you almost never want to `get()` without calling `isPresent()`, so `map()` is usually a better way to express this.

Reply:
Got it, thanks

Reply:
Done. I ended up using `Optional.ifPresent()`, because the lambda doesn't return anything.